### PR TITLE
Feature/notify message

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #MQvsU9co](https://trello.com/c/MQvsU9co/12-3-as-a-user-i-should-be-able-to-receive-notifications-when-someone-sends-me-a-message-so-i-can-be-aware-of-it)
---
#### Description:

* Notify a user when he receives a message
---
#### Notes:
*
---
#### Tasks:
- [x] Send notification
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: app/services/api/v1/message_service.rb [hunk 1]
```diff
@@ -11,7 +11,20 @@ def create
         conversation = Conversation.find(@conversation_id)
         raise InvalidConversationError unless conversation.user1_id == @current_user.id || conversation.user2_id == @current_user.id
 
-        Message.create!(content: @message_params[:content], user: @current_user, conversation_id: @conversation_id)
+        @message = Message.create!(content: @message_params[:content], user: @current_user, conversation_id: @conversation_id)
+        notify
```
Comment (enzofab91) [inline app/services/api/v1/message_service.rb]: You can check if message is saved before sending the notification.

Something like:

```suggestion
        return unless @message.save?

        notify
```

File: app/services/api/v1/message_service.rb [hunk 2]
```diff
@@ -11,7 +11,20 @@ def create
...
+        @message
+      end
+
+      def notify
+        conversation = Conversation.find(@conversation_id)
+        user1 = User.find(conversation.user1_id)
+        user2 = User.find(conversation.user2_id)
```
Comment (enzofab91) [inline app/services/api/v1/message_service.rb]: You can wrap this logic into a method in `Conversation` model and here just call it.

```suggestion
        user  = Conversation.recipient_user
```

**Review by enzofab91 — COMMENTED** [review]
Nice! Left some comments.

For testing purposes this gem can be useful: https://github.com/ryanb/letter_opener

File: app/services/api/v1/message_service.rb [hunk 3]
```diff
@@ -11,7 +11,15 @@ def create
...
         raise InvalidConversationError unless conversation.user1_id == @current_user.id || conversation.user2_id == @current_user.id
 
-        Message.create!(content: @message_params[:content], user: @current_user, conversation_id: @conversation_id)
+        @message = Message.create!(content: @message_params[:content], user: @current_user, conversation_id: @conversation_id)
+        notify if @message.persisted?
+        @message
+      end
+
+      def notify
+        conversation = Conversation.find(@conversation_id)
```
Comment (jpascual1994) [inline app/services/api/v1/message_service.rb]: Maybe you can pass the conversation as a parameter so you don't have to find it again.
Also, I think this method could be a private method

**Review by juanmferreira93 — APPROVED** [review]

**Review by enzofab91 — APPROVED** [review]
🙌

**Pull request merged**